pkg/controller/backup_controller.go

```diff
@@ -80,30 +79,28 @@ func NewBackupController(
 	backupInformer informers.BackupInformer,
 	client arkv1client.BackupsGetter,
 	backupper backup.Backupper,
-	objectStoreConfig api.CloudProviderConfig,
-	bucket string,
 	pvProviderExists bool,
 	logger logrus.FieldLogger,
 	logLevel logrus.Level,
 	pluginRegistry plugin.Registry,
 	backupTracker BackupTracker,
+	backupLocationInformer informers.BackupStorageLocationInformer,
 	metrics *metrics.ServerMetrics,
 ) Interface {
 	c := &backupController{
-		backupper:         backupper,
-		objectStoreConfig: objectStoreConfig,
-		bucket:            bucket,
-		pvProviderExists:  pvProviderExists,
-		lister:            backupInformer.Lister(),
-		listerSynced:      backupInformer.Informer().HasSynced,
-		client:            client,
-		queue:             workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "backup"),
-		clock:             &clock.RealClock{},
-		logger:            logger,
-		logLevel:          logLevel,
-		pluginRegistry:    pluginRegistry,
-		backupTracker:     backupTracker,
-		metrics:           metrics,
+		backupper:            backupper,
```

Comment:
it would be nice to convert the controllers to use the `genericController` type while we're in there modifying them.  This requires:

- embedding a `*genericController` in the `backupController`
- initializing it within the `NewBackupController` func using the `newGenericController` func
- removing the `syncHandler` field from the `backupController` (`genericController` has its own) 
- setting the `cacheSyncWaiters` field in the constructor (removes the need to store each of them as their own fields)
- removing the `queue` field
- removing the `Run`, `runWorker`, and `processNextWorkItem` funcs

I did the same for the download request controller -- see https://github.com/heptio/ark/pull/769/files#diff-cdd2d4325fc0100b6d5a4039fc99792e -- although I think I missed at least one step now that I'm testing.

Reply:
If you want to defer this to a separate PR I'm ok with that, but let's get it done for 0.10

Reply:
I hadn't looked at it yet; want me to push the other fixes then see where we're at?

Reply:
sure